In [1]:
##import library
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
##load dataset
df = pd.read_csv("Breast_Cancer.csv")
df.head()

Age   Race Marital Status T Stage  N Stage 6th Stage  \
0   68  White        Married       T1      N1       IIA   
1   50  White        Married       T2      N2      IIIA   
2   58  White       Divorced       T3      N3      IIIC   
3   58  White        Married       T1      N1       IIA   
4   47  White        Married       T2      N1       IIB   

               differentiate Grade   A Stage  Tumor Size Estrogen Status  \
0      Poorly differentiated     3  Regional           4        Positive   
1  Moderately differentiated     2  Regional          35        Positive   
2  Moderately differentiated     2  Regional          63        Positive   
3      Poorly differentiated     3  Regional          18        Positive   
4      Poorly differentiated     3  Regional          41        Positive   

  Progesterone Status  Regional Node Examined  Reginol Node Positive  \
0            Positive                      24                      1   
1            Positive                      14                      5   
2            Positive                      14                      7   
3            Positive                       2                      1   
4            Positive                       3                      1   

   Survival Months Status  
0               60  Alive  
1               62  Alive  
2               75  Alive  
3               84  Alive  
4               50  Alive

In [5]:
df.columns

Index(['Age', 'Race', 'Marital Status', 'T Stage ', 'N Stage', '6th Stage',
       'differentiate', 'Grade', 'A Stage', 'Tumor Size', 'Estrogen Status',
       'Progesterone Status', 'Regional Node Examined',
       'Reginol Node Positive', 'Survival Months', 'Status'],
      dtype='object')

In [8]:
df.shape

(4024, 16)

In [3]:
##data description
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4024 entries, 0 to 4023
Data columns (total 16 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Age                     4024 non-null   int64 
 1   Race                    4024 non-null   object
 2   Marital Status          4024 non-null   object
 3   T Stage                 4024 non-null   object
 4   N Stage                 4024 non-null   object
 5   6th Stage               4024 non-null   object
 6   differentiate           4024 non-null   object
 7   Grade                   4024 non-null   object
 8   A Stage                 4024 non-null   object
 9   Tumor Size              4024 non-null   int64 
 10  Estrogen Status         4024 non-null   object
 11  Progesterone Status     4024 non-null   object
 12  Regional Node Examined  4024 non-null   int64 
 13  Reginol Node Positive   4024 non-null   int64 
 14  Survival Months         4024 non-null   int64 
 15  Stat

The breast cancer dataset consists of 4,024 instances and 16 variables including the survival status. Among these, Age, Tumor Size, Regional Nodes Examined, Regional Nodes Positive, and Survival Months are numerical variables, while the remaining variables are categorical.

In [6]:
##checking missing value
df.isnull().sum()

Age                       0
Race                      0
Marital Status            0
T Stage                   0
N Stage                   0
6th Stage                 0
differentiate             0
Grade                     0
A Stage                   0
Tumor Size                0
Estrogen Status           0
Progesterone Status       0
Regional Node Examined    0
Reginol Node Positive     0
Survival Months           0
Status                    0
dtype: int64

There is no missing value in the dataset. 

In [10]:
##checking duplicate value
df.duplicated().sum()

1

The dataset contains only one duplicate value. 

In [11]:
##remove duplicate value
df.drop_duplicates(inplace=True)

In [12]:
##numerical variables summary 
df.describe()

Age   Tumor Size  Regional Node Examined  \
count  4023.000000  4023.000000             4023.000000   
mean     53.969923    30.477007               14.358439   
std       8.963118    21.121253                8.100241   
min      30.000000     1.000000                1.000000   
25%      47.000000    16.000000                9.000000   
50%      54.000000    25.000000               14.000000   
75%      61.000000    38.000000               19.000000   
max      69.000000   140.000000               61.000000   

       Reginol Node Positive  Survival Months  
count            4023.000000      4023.000000  
mean                4.158837        71.301765  
std                 5.109724        22.923009  
min                 1.000000         1.000000  
25%                 1.000000        56.000000  
50%                 2.000000        73.000000  
75%                 5.000000        90.000000  
max                46.000000       107.000000

In [13]:
##categorical variables summary
df.describe(include= 'object')

Race Marital Status T Stage  N Stage 6th Stage  \
count    4023           4023     4023    4023      4023   
unique      3              5        4       3         5   
top     White        Married       T2      N1       IIA   
freq     3412           2642     1786    2731      1304   

                    differentiate Grade   A Stage Estrogen Status  \
count                        4023  4023      4023            4023   
unique                          4     4         2               2   
top     Moderately differentiated     2  Regional        Positive   
freq                         2350  2350      3931            3754   

       Progesterone Status Status  
count                 4023   4023  
unique                   2      2  
top               Positive  Alive  
freq                  3325   3407